atsign-foundation/at_server
# fix: Commit log special key bug

Fixes https://github.com/atsign-foundation/at_server/issues/2114

**- What I did**
- fixed a bug in _specialKey method in commit log. Due to this bug, lastCommitSequenceNumber was incorrectly returned when regex is passed.
**- How I did it**
- modified a condition to check public encryption key from "public:" to "public:publickey"
- match only reserved shared key (`@bob:shared_key@alice`, `shared_key.bob@alice`)  in _specialKey
- removed pkam signature and signing private key from _specialKey check
- added unit tests for getEntries and lastSequenceNumber
- fixed an issue in commit log test by replacing commitLog.getBox().add(doesn't update commitLog cache) with commitLog.add
- renamed method names to reflect their purpose
**- How to verify it**
- tests should pass
https://github.com/atsign-foundation/at_client_sdk/pull/1414

Dependent at_commons PR:
https://github.com/atsign-foundation/at_libraries/pull/694
